sklearn/utils/fixes.py

```diff
@@ -107,22 +109,39 @@ def _eigh(*args, **kwargs):
 
 
 # remove when https://github.com/joblib/joblib/issues/1071 is fixed
-def delayed(function):
+def delayed(function, config=None):
```

Comment:
I think the original bug is big enough for us to have a private `_delayed(function, config)` so it can be backported to `1.2.1` to avoid new API in a bug fix release. I'll be okay if this is done in separate PR.

Unfortunately, developers using `joblib`, will need to update their code to work correctly with globally setting `transform="pandas"`. Moreover, developers will need to depend on `utils.fixes.delayed` for a while. This suggests to me that we need to render the docs for `utils.fixes.delayed` and properly document it.